lib/lrama/grammar/symbols/resolver.rb

```diff
@@ -44,7 +44,9 @@ def add_term(id:, alias_name: nil, tag: nil, token_id: nil, replace: false)
         end
 
         def add_nterm(id:, alias_name: nil, tag: nil)
-          return if find_symbol_by_id(id)
+          if (sym = find_symbol_by_id(id))
+            return sym
```

Comment:
I was wondering why you changed it to return `sym`, but I see it's because the previous version could return `nil`, which would fail type checking. Understood. It also aligns the behavior with `add_term`, which seems good.